### PR TITLE
Add spec for verify json output

### PR DESCRIPTION
**Pull request by byronchien**
Adds spec for `notation verify $IMAGE --output json`.

Changes from previous iteration:
- adds a details field for additional information related to the result
- outputs a JSON object on failure and skipped results

Related issues and PRs:
- https://github.com/notaryproject/notation/pull/527
- https://github.com/notaryproject/notation/issues/546
- https://github.com/notaryproject/notation/issues/547

### CODE REVIEW

**Comment by FeynmanZhou** [comment]
@byronchien  You may need to rebase your branch and sync with the upstream main as it has conflicts.

File: specs/commandline/verify.md [hunk 1]
```diff
@@ -197,3 +197,47 @@ export NOTATION_EXPERIMENTAL=1
...
+```
+
+An example of output messages for a successful verification:
+
+```text
+{
+    "reference": "localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9",
+    "userMetadata": {
+        "io.wabbit-networks.buildId": "123"
+    },
```
Comment (priteshbandi) [inline specs/commandline/verify.md:216]: Should we also add signature reference/digest? And move userMetadata to be a child of it?
```
{
    "reference": "localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9",
    "signatures":  [
        {
            "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a33",
            "userDefinedAttributes":  {
                "io.wabbit-networks.buildId": "123" 
            }
....
```

Also to be consistent with [inspect](https://github.com/notaryproject/notation/blob/main/specs/commandline/inspect.md#inspect-signatures-on-the-supplied-oci-artifact-with-an-example-of-json-output), should we rename `userMetadata` to `userDefinedAttributes`

**Review by priteshbandi — DISMISSED** [review]
LGTM

File: specs/commandline/verify.md [hunk 2]
```diff
@@ -197,3 +197,39 @@ export NOTATION_EXPERIMENTAL=1
...
+
+Use the `--output` flag to format verification output in json.
+
+```shell
+notation verify --output json localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+An example of output messages for a successful verification:
+
+```text
```
Comment (priteshbandi) [inline specs/commandline/verify.md:211]:
```suggestion
```json
```

File: specs/commandline/verify.md [hunk 3]
```diff
@@ -197,3 +197,39 @@ export NOTATION_EXPERIMENTAL=1
...
+            }
+        }
+    ],
+    "result": "success"
+}
+```
+
+An example of output messages for a verification skipped by the trust policy:
+
+```text
```
Comment (priteshbandi) [inline specs/commandline/verify.md:228]:
```suggestion
```json
```

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Comment by sajayantony** [comment]
+1. I think it's worth merging this. Typically it would be good to call out all possible values of `result` if there are more.

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Comment by shizhMSFT** [comment]
Should it be merged post v1?

**Comment by yizha1** [comment]
> Should it be merged post v1?

+1 on merging it post v1.

File: specs/commandline/verify.md [hunk 4]
```diff
@@ -197,3 +197,39 @@ export NOTATION_EXPERIMENTAL=1
 # The value of --scope should be set base on the trust policy configuration
 notation verify --oci-layout --scope "local/hello-world" hello-world:v1
 ```
+
+### Verify signatures on an OCI artifact with json output
+
+Use the `--output` flag to format verification output in json.
```
Comment (Two-Hearts) [inline specs/commandline/verify.md:203]: Can you also add this new flag along with its description into the `Outline` session of this spec? (https://github.com/notaryproject/notation/blob/main/specs/commandline/verify.md#outline)

File: specs/commandline/verify.md [hunk 5]
```diff
@@ -197,3 +197,39 @@ export NOTATION_EXPERIMENTAL=1
...
+
+```shell
+notation verify --output json localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
```
Comment (Two-Hearts) [inline specs/commandline/verify.md:206]: Just curious, what error message are we expecting from the following command:
`notation verify --output <something_not_defined> localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9`

**Review by yizha1 — CHANGES_REQUESTED** [review]
@byronchien I reviewed this PR again. Here are two comments:
- Unified solution for all Notation CLI commands: I couldn't recall any reason behind this PR that only verify command is addressed. The Json output should be considered for all other Notation CLI commands. So, it is suggested to create a unified solution for all Notation CLI commands.
- About the flag name `-o --output`: There is a new feature about signing and verifying arbitrary files. For this feature, we may need to output signature to a file during signing, there may be a conflict of using `--output`. It seems `notation inspect` command already uses the flag `--output`. We need to align the usage of flag `--output`.
/cc: @priteshbandi @shizhMSFT @FeynmanZhou

**Pull request closed without merging**